plugins/chat/plugin.rb

```diff
@@ -303,32 +304,29 @@ module ::Chat
   end
 
   register_presence_channel_prefix("chat") do |channel_name|
-    next nil unless channel_name == "/chat/online"
-    config = PresenceChannel::Config.new
-    config.allowed_group_ids = Chat.allowed_group_ids
-    config
+    next if channel_name != "/chat/online"
+    PresenceChannel::Config.new.tap { |config| config.allowed_group_ids = Chat.allowed_group_ids }
   end
 
   register_presence_channel_prefix("chat-reply") do |channel_name|
-    if (
-         channel_id, thread_id =
-           channel_name.match(%r{^/chat-reply/(\d+)(?:/thread/(\d+))?$})&.captures
-       )
-      chat_channel = nil
-      if thread_id
-        chat_channel = Chat::Thread.find_by!(id: thread_id, channel_id: channel_id).channel
+    channel_id, thread_id = Chat::PRESENCE_REGEXP.match(channel_name)&.captures
+
+    next if channel_id.blank?
+
+    chat_channel =
+      if thread_id.present?
+        Chat::Thread.find_by(id: thread_id, channel_id:)&.channel
       else
-        chat_channel = Chat::Channel.find(channel_id)
+        Chat::Channel.find_by(id: channel_id)
       end
 
-      PresenceChannel::Config.new.tap do |config|
-        config.allowed_group_ids = chat_channel.allowed_group_ids
-        config.allowed_user_ids = chat_channel.allowed_user_ids
-        config.public = !chat_channel.read_restricted?
-      end
+    next if chat_channel.nil?
```

Comment:
This line is the actual fix. It ensures we always have a `chat_channel` before "using" it.